server/build_event_protocol/build_event_handler/build_event_handler.go

```diff
@@ -789,6 +791,63 @@ func extractOptionsFromStartedBuildEvent(event *build_event_stream.BuildEvent) (
 	return "", nil
 }
 
+func getCursesLinesFromStartedBuildEvent(event *build_event_stream.BuildEvent) int {
+	// the number of lines curses can overwrite is 2 + the ui_actions shown:
+	// 1 for the progress tracker, 1 for each action, and 1 blank line
+	options, err := extractOptionsFromStartedBuildEvent(event)
+	if err != nil {
+		log.Errorf("Could not extract options for ui_actions_show, defaulting to 8: %v", err)
+		return 10
+	}
+	optionsList, err := shlex.Split(options)
+	if err != nil {
+		log.Errorf("Could not shlex split options for ui_actions_show, defaulting to 8: %v", err)
+		return 10
+	}
+	actionsShownValues := getValuesForOptionNameFromOptionsList(optionsList, "ui_actions_shown")
+	cursesValues := getValuesForOptionNameFromOptionsList(optionsList, "ui_actions_shown")
+	if len(cursesValues) > 1 {
+		log.Errorf("Too many arguments to curses, assuming auto: %v", cursesValues)
+	}
+	if len(cursesValues) == 1 && cursesValues[0] == "no" {
+		return 0
+	}
+	if len(actionsShownValues) > 1 {
+		log.Errorf("Too many arguments to ui_actions_shown, defaulting to 8: %v", actionsShownValues)
+	}
+	if len(actionsShownValues) == 1 {
+		n, err := strconv.Atoi(actionsShownValues[0])
+		if err != nil {
+			log.Errorf("Invalid argument to ui_actions_shown, defaulting to 8: %v", err)
+		} else if n < 1 {
+			return 3
+		} else {
+			return n + 2
+		}
+	}
+	return 10
+}
+
+func getValuesForOptionNameFromOptionsList(optionsList []string, value string) []string {
```

Comment:
Should `value` be `name` or `option` or `optionName`?

Reply:
I think `value` should be `optionName`? That seems clearest. I don't think `option` should be `optionName` because `option` includes the preceding `--` as well as the subsequent `=<value>`, so I think it's more than the name. I've adjusted the PR to reflect this, let me know what you think.